SPLIT_USB_DETECT prevents keyboard detection on cold boot

akio-ito opened the pull request:
I was having problems when turning on the computer (Mac M1) and when I needed to wake up from sleep mode.
Apparently it fixed the problem.

References:
https://github.com/sunaku/vial-qmk/tree/sunaku_remnant/keyboards/ergohaven/remnant/keymaps/sunaku

https://github.com/qmk/qmk_firmware/issues/8990

https://github.com/qmk/qmk_firmware/issues/18511

akio-ito committed:
SPLIT_USB_DETECT prevents keyboard detection on cold boot

References:
https://github.com/sunaku/vial-qmk/tree/sunaku_remnant/keyboards/ergohaven/remnant/keymaps/sunaku

https://github.com/qmk/qmk_firmware/issues/8990

https://github.com/qmk/qmk_firmware/issues/18511

sunaku commented on keyboards/cyboard/dactyl/keymaps/singlearc_number_row/rules.mk:
```diff
@@ -6,3 +6,16 @@ TAP_DANCE_ENABLE = yes
+#-----------------------------------------------------------------------------
+
+# https://docs.qmk.fm/#/feature_caps_word?id=caps-word
+CAPS_WORD_ENABLE = yes
```
This isn't needed for the `SPLIT_USB_DETECT` fix. :policeman: This just enables [Pascal Getreuer's CapsWord feature](https://getreuer.info/posts/keyboards/caps-word/index.html).

sunaku commented on keyboards/cyboard/dactyl/keymaps/singlearc_number_row/rules.mk:
```diff
@@ -6,3 +6,16 @@ TAP_DANCE_ENABLE = yes
+CAPS_WORD_ENABLE = yes
+
+# https://docs.qmk.fm/#/custom_quantum_functions?id=deferred-execution
+DEFERRED_EXEC_ENABLE = yes
```
This isn't needed for the `SPLIT_USB_DETECT` fix. :policeman: This just enables QMK deferred execution for [my home row mods enhancement](https://github.com/manna-harbour/qmk_firmware/pull/56).

akio-ito committed:
This isn't needed for the SPLIT_USB_DETECT fix.

https://github.com/Cyboard-DigitalTailor/vial-qmk/pull/1

akio-ito reviewed (commented):
Fixed. 
Sorry, I'm not very used to PR.
In practical terms, it might be better to adjust to the level above.
keyboards/dactyl/config.h
keyboards/dactyl/rules.mk

https://github.com/akio-ito/vial-qmk/commit/c89dd5b639771d95f23dd5fec18c38461a692e83

sunaku/Eric, feel free to close this PR without merging.

akio-ito committed:
SPLIT_USB_DETECT prevents keyboard detection on cold boot, for all boards.

https://github.com/Cyboard-DigitalTailor/vial-qmk/pull/1#pullrequestreview-1357898907
